base/utils/vmaas_test.go

```diff
@@ -0,0 +1,245 @@
+
+	assert.Equal(t, 7, len(merged.GetAvailableUpdates()))
+	assert.Equal(t, true, updatesAscending(merged.GetAvailableUpdates()))
+}
+
+func TestMergeUpdatesWithAGreaterLonger(t *testing.T) {
+	var listA vmaas.UpdatesV2ResponseUpdateList
+	var listB vmaas.UpdatesV2ResponseUpdateList
+
+	err := json.Unmarshal(updatesBash9, &listA)
+	assert.Nil(t, err)
+	err = json.Unmarshal(updatesBash10, &listB)
+	assert.Nil(t, err)
+
+	merged, err := mergeUpdates(listA, listB)
+	assert.Nil(t, err)
+
+	assert.Equal(t, 7, len(merged.GetAvailableUpdates()))
+	assert.Equal(t, true, updatesAscending(merged.GetAvailableUpdates()))
+}
+
+func TestMergeUpdatesWithALesserShorter(t *testing.T) {
+	var listA vmaas.UpdatesV2ResponseUpdateList
+	var listB vmaas.UpdatesV2ResponseUpdateList
+
+	err := json.Unmarshal(updatesBash11, &listA)
+	assert.Nil(t, err)
+	err = json.Unmarshal(updatesBash12, &listB)
+	assert.Nil(t, err)
+
+	merged, err := mergeUpdates(listA, listB)
+	assert.Nil(t, err)
+
+	assert.Equal(t, 8, len(merged.GetAvailableUpdates()))
+	assert.Equal(t, true, updatesAscending(merged.GetAvailableUpdates()))
+}
+
+func TestMergeUpdatesWithAGreaterShorter(t *testing.T) {
+	var listA vmaas.UpdatesV2ResponseUpdateList
+	var listB vmaas.UpdatesV2ResponseUpdateList
+
+	err := json.Unmarshal(updatesBash8, &listA)
+	assert.Nil(t, err)
+	err = json.Unmarshal(updatesBash7, &listB)
+	assert.Nil(t, err)
+
+	merged, err := mergeUpdates(listA, listB)
+	assert.Nil(t, err)
+
+	assert.Equal(t, 7, len(merged.GetAvailableUpdates()))
+	assert.Equal(t, true, updatesAscending(merged.GetAvailableUpdates()))
+}
+
+func TestMergeUpdatesWithDifferentRepos(t *testing.T) {
+	var listA vmaas.UpdatesV2ResponseUpdateList
+	var listB vmaas.UpdatesV2ResponseUpdateList
+
+	err := json.Unmarshal(updatesBash13, &listA)
+	assert.Nil(t, err)
+	err = json.Unmarshal(updatesBash14, &listB)
+	assert.Nil(t, err)
+
+	merged, err := mergeUpdates(listA, listB)
+	assert.Nil(t, err)
+
+	assert.Equal(t, 10, len(merged.GetAvailableUpdates()))
+	assert.Equal(t, true, updatesAscending(merged.GetAvailableUpdates()))
+}
+
+func updatesAscending(updates []vmaas.UpdatesV2ResponseAvailableUpdates) bool {
+	var prev string
+	var prevRepo string
+
+	for i, u := range updates {
+		nevra, err := ParseNevra(u.GetPackage())
+		if err != nil {
+			return false
+		}
+
+		if i == 0 {
+			prev = nevra.Version
+			prevRepo = *u.Repository
+			continue
+		}
+		// Simplified, ideally should be map check.
+		if prev >= nevra.Version && prevRepo == *u.Repository {
+			return false
+		}
+	}
+
+	return true
+}
+
+func TestMergeVMaaSResponses(t *testing.T) {
+	var respA vmaas.UpdatesV2Response
+	var respB vmaas.UpdatesV2Response
+
+	err := json.Unmarshal(kernel3101, &respA)
+	assert.Nil(t, err)
+	err = json.Unmarshal(kernel3102, &respB)
+	assert.Nil(t, err)
+
+	res, err := MergeVMaaSResponses(&respA, &respB)
+	assert.Nil(t, err)
+
+	updateList := (*res.UpdateList)[kernel310Pkg]
+	assert.Equal(t, 6, len(updateList.GetAvailableUpdates()))
```

Comment:
There's a bug. Merging `kernel3101` and `kernel3102` should give you 7 different updates.

Reply:
These versions are unique `10.1, 10.4, 10.7, 10.9, 11.0, 12.8`. I see 6.

Reply:
@MichaelMraka do you confirm, or there indeed is something there?

Reply:
The  `10.9` version have different errata so it should be twice in the list. Updates are not completely equal.

Reply:
I made up this data, do you think it will happen?

Reply:
Yes, it actually can happen. E.g. java package is in more repos (one for base RHEL, other for layered products) and updates go to all of them but as a different errata.